### ISSUE TITLE
Feature: Allow execution of scripts w/ concatenated binary 

### ISSUE THREAD

**Issue by sambacha** [direnv/direnv#1034]
##  Run shell scripts having concatenated binary content.[^1]

[POSIX was recently revised to require this behavior](https://pubs.opengroup.org/onlinepubs/9699919799/):

>"The input file may be of any type, but the initial portion of the
> file intended to be parsed according to the shell grammar (XREF to
> XSH 2.10.2 Shell Grammar Rules) shall consist of characters and
> shall not contain the NUL character. The shell shall not enforce
> any line length limits."

{ ... }

>"Earlier versions of this standard required that input files to the
>shell be text files except that line lengths were unlimited.
>However, that was overly restrictive in relation to the fact that
>shells can parse a script without a trailing newline, and in
>relation to a common practice of concatenating a shell script
>ending with an 'exit' or 'exec $command' with a binary data payload
>to form a single-file self-extracting archive[^2]."

## POSIX me harder
TLDR:  POSIX says we shall allow execution of scripts with concatenated binary and suggests checking a line exists before the first NUL character with a lowercase letter or expansion. This would for example, allow for go to be used as a scripting language for shell scripts.[^3]

### References

[^1]: https://justine.lol/cosmopolitan/index.html
[^2]: http://austingroupbugs.net/view.php?id=1250
[^3]: http://austingroupbugs.net/view.php?id=1226#c4394
[^4]:  https://github.com/freebsd/freebsd-src/commit/9a1cd363318b7e9e70ef6af27d1675b371c16b1a

Initial source: https://github.com/jart/zsh/commit/94a4bc14bb2e415ec3d10cf716512bd3e0d99f48
